src/imap.rs

```diff
@@ -292,20 +336,49 @@ impl Imap {
         }
     }
 
-    /// Connects or reconnects if not already connected.
+    /// Determine server capabilities if not done yet.
+    async fn determine_capabilities(&mut self) -> Result<()> {
+        if self.capabilities_determined {
+            return Ok(());
+        }
+
+        match &mut self.session {
+            Some(ref mut session) => match session.capabilities().await {
+                Ok(caps) => {
+                    self.config.can_idle = caps.has_str("IDLE");
+                    self.config.can_move = caps.has_str("MOVE");
+                    self.capabilities_determined = true;
+                    Ok(())
+                }
+                Err(err) => {
+                    bail!("CAPABILITY command error: {}", err);
+                }
+            },
+            None => {
+                bail!("Can't determine server capabilities because connection was not established")
+            }
+        }
+    }
+
+    /// Prepare for IMAP operation.
     ///
-    /// This function emits network error if it fails.  It should not
-    /// be used during configuration to avoid showing failed attempt
-    /// errors to the user.
-    async fn setup_handle(&mut self, context: &Context) -> Result<()> {
-        let res = self.try_setup_handle(context).await;
+    /// Ensure that IMAP client is connected, folders are created and IMAP capabilities are
+    /// determined.
+    ///
+    /// This function emits network error if it fails.  It should not be used during configuration
+    /// to avoid showing failed attempt errors to the user.
```

Comment:
BTW network errors are to be removed in https://github.com/deltachat/deltachat-core-rust/pull/2319